domain/application/service/migration.go

```diff
@@ -0,0 +1,243 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
```

Comment:
Shouldn't most of these methods go into `domain/application/modelmigration`? 
We even already have Charm retrieval methods there so you don't have to duplicate here.

Reply:
They do different things, also I don't want to bring in the whole application service. For example, the import service has a different way to access types (applications and charms) that can be optimised for migration.